File: paddler_types/src/kv_cache_dtype.rs
use serde::Deserialize;
use serde::Serialize;

#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[expect(
    non_camel_case_types,
    reason = "variant names mirror ggml type identifiers (e.g. GGML_TYPE_IQ4_NL) for parity with llama.cpp's --cache-type-k/-v"
)]
pub enum KvCacheDtype {
    F32,
    F16,
    BF16,
    Q8_0,
    Q4_0,
    Q4_1,
    IQ4_NL,
    Q5_0,
    Q5_1,
}

File: paddler_client_python/paddler_client/kv_cache_dtype.py
from enum import StrEnum


class KvCacheDtype(StrEnum):
    F32 = "F32"
    F16 = "F16"
    BF16 = "BF16"
    Q8_0 = "Q8_0"
    Q4_0 = "Q4_0"
    Q4_1 = "Q4_1"
    IQ4_NL = "IQ4_NL"
    Q5_0 = "Q5_0"
    Q5_1 = "Q5_1"
